pkg/repository/workspace_redis.go

```diff
@@ -51,3 +55,46 @@ func (wr *WorkspaceRedisRepository) SetConcurrencyLimitByWorkspaceId(workspaceId
 
 	return nil
 }
+
+type AuthInfo struct {
+	Workspace *types.Workspace
+	Token     *types.Token
+}
+
+func (wr *WorkspaceRedisRepository) AuthorizeToken(token string) (*types.Token, *types.Workspace, error) {
+	tokenKey := common.RedisKeys.WorkspaceAuthorizedToken(token)
+	res, err := wr.rdb.Get(context.Background(), tokenKey).Result()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if res == "" {
+		return nil, nil, errors.New("token not found")
+	}
+
+	info := &AuthInfo{}
+	err = json.Unmarshal([]byte(res), &info)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return info.Token, info.Workspace, nil
+}
+
+func (wr *WorkspaceRedisRepository) CacheAuthorizationToken(token *types.Token, workspace *types.Workspace) error {
```

Comment:
A nitpick, but can we say `Set` instead of `Cache` to keep with the naming convention of the other redis repo related functions?